src/node_crypto.cc

```diff
@@ -2192,6 +2191,39 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
 }
 
 
+static void ParseX509(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsArrayBufferView());
+  ArrayBufferViewContents<unsigned char> buf(args[0].As<ArrayBufferView>());
+  const unsigned char* data = buf.data();
+  unsigned data_len = buf.length();
```

Comment:
Couple of width/sign issues here: `buf.length()` returns a `size_t`, `d2i_X509` takes an `int`. Suggestion:
```c++
size_t data_len = buf.length();
CHECK_LE(data_len, INT_MAX);
```